src/rust/engine/fs/store/src/local.rs

```diff
@@ -395,26 +553,36 @@ impl ByteStore {
       return Ok(Some(f(&[])));
     }
 
-    let dbs = match entry_type {
-      EntryType::Directory => self.inner.directory_dbs.clone(),
-      EntryType::File => self.inner.file_dbs.clone(),
-    }?;
-    let res = dbs
-      .load_bytes_with(digest.hash, move |bytes| {
-        if bytes.len() == digest.size_bytes {
-          Ok(f(bytes))
-        } else {
-          Err(format!(
-            "Got hash collision reading from store - digest {:?} was requested, but retrieved \
-                bytes with that fingerprint had length {}. Congratulations, you may have broken \
-                sha256! Underlying bytes: {:?}",
-            digest,
-            bytes.len(),
-            bytes
-          ))
-        }
-      })
-      .await;
+    let res = if ByteStore::uses_large_file_store(digest.size_bytes) {
+      let src = &self.large_file_path(digest);
+      let mut reader = std::fs::File::open(src)
+        .map_err(|e| format!("Failed to read {src:?} for digest {digest:?}: {e}"))?;
+      let mut writer: Vec<u8> = vec![];
+      io::copy(&mut reader, &mut writer)
+        .map_err(|e| format!("Failed to load large file into memory: {e}"))?;
+      Ok(Some(f(&writer[..])))
```

Comment:
This will always need to exist (as opposed to forcing the caller to symlink) chiefly for the case that we're materializing a file that _shouldn't_ be immutable. We'll need to copy no matter what.